src/widgets/immunizations/vaccination-row.component.tsx

```diff
@@ -0,0 +1,194 @@
+                      ? toggleOpen
+                        ? "#omrs-icon-chevron-up"
+                        : "#omrs-icon-chevron-down"
+                      : ""
+                  }
+                />
+              </svg>
+            </div>
+            <span>{patientImmunization.vaccineName}</span>
+          </td>
+          <td>
+            <div className={`${styles.alignRight}`}>
+              {getRecentVaccinationText(t, patientImmunization)}
+            </div>
+          </td>
+          <td>
+            <div className={styles.headerAdd}>
+              <button
+                className={`${styles.addButton}`}
+                onClick={() =>
+                  openWorkspaceTab(ImmunizationsForm, "Immunizations Form", [
+                    {
+                      vaccineName: patientImmunization?.vaccineName,
+                      vaccineUuid: patientImmunization?.vaccineUuid,
+                      sequences: patientImmunization?.sequences
+                    }
+                  ])
+                }
+              >
+                +
+              </button>{" "}
+            </div>
+          </td>
+        </tr>
+        {toggleOpen && (
+          <tr
+            id={patientImmunization?.uuid}
+            className={`immunizationSequenceRow ${vaccinationRowStyles.sequenceRow}`}
+          >
+            <td colSpan={4}>
+              <table
+                className={`omrs-type-body-regular immunizationSequenceTable ${vaccinationRowStyles.sequenceTable}`}
+              >
+                <thead>
+                  <tr>
+                    <td>{capitalize(t("sequence", "Sequence"))}</td>
+                    <td>
+                      {capitalize(t("vaccination date", "Vaccination Date"))}
+                    </td>
+                    <td>
+                      {capitalize(t("expiration date", "Expiration Date"))}
+                    </td>
+                    <td></td>
+                  </tr>
+                </thead>
+                <tbody>
+                  {renderSequenceTable(match, t, patientImmunization)}
+                </tbody>
+              </table>
+            </td>
+          </tr>
+        )}
+      </React.Fragment>
+    )
+  );
+}
+
+function getRecentVaccinationText(t, patientImmunization) {
+  if (!hasExistingDoses(patientImmunization)) {
+    return "";
+  }
+  let recentDose = patientImmunization.doses[0];
+  if (hasSequence(patientImmunization)) {
+    return (
+      recentDose.sequenceLabel +
+      " on " +
+      dayjs(recentDose.occurrenceDateTime).format("DD-MMM-YYYY")
+    );
+  }
+  const singleDoseText = t("single dose", "Single Dose");
+  return (
+    singleDoseText +
+    " on " +
+    dayjs(recentDose.occurrenceDateTime).format("DD-MMM-YYYY")
+  );
+}
+
+function hasExistingDoses(patientImmunization: any) {
+  return patientImmunization.doses && patientImmunization.doses.length > 0;
+}
+
+function hasSequence(patientImmunization) {
```

Comment:
What is `patientImmuninization`?

Reply:
Would provide a type for `patientImmuninization`.